### ISSUE TITLE
Initiative module does not handle empty definition parameters

### ISSUE THREAD

**Issue by gettek** [gettek/terraform-azurerm-policy-as-code#46]
Some policy definitions (Built-In or Custom) may have empty parameters which the initiative module does not handle correctly

See PR #45

**Comment by toby-p9** [comment]
Hi there, first of all: thanks for fixing this! The fix works for policy definitions.
However, I still have similar issues in the following two cases:
1) when assigning a policy initiative that has parameters, but you don't assign any (so the initiative will use defaults only. Prime example: the Azure Security Benchmark). This results in the following change at every apply:
![screen_1](https://user-images.githubusercontent.com/72439212/193015174-8b6c4977-ec20-4ad2-832f-18a2180d313b.jpg)
Workaround: take a random parameter and assign the default value. This will apply the parameter, change nothing but prevent the change at every Terraform apply.
2) when including policies that do not have any parameters in an initiative. This will happen:
![screen_2](https://user-images.githubusercontent.com/72439212/193027769-1282df51-fad5-4899-bf7f-57d19b44cd10.jpg)
Workaround: I modified line 23 in modules > initiative > main.tf and included a length check. Instead of 
`parameter_values = jsonencode({`
I wrote
`parameter_values = (length(policy_definition_reference.value.parameters) == 0) ? "" : jsonencode({`


Both of these seem like provider issues at heart, [similar to this one](https://github.com/hashicorp/terraform-provider-azurerm/issues/5500).
My workarounds work, but you might be able to handle them in a better way in case you are interested. I just required a quick & dirty fix...
